src/sc.sh

```diff
@@ -1,18 +1,12 @@
-name(){
-	basename `realpath .`
-}
+alias t="basename \$(realpath .)" # Assumption: prints an alphanumeric string
 b(){
-	flags=-O2\ -fsanitize=address
-	[ a"$1" = a-d ] && flags=-g
-	g++ -std=c++11 -lm -Wall -Wno-sign-compare -Wshadow $flags "$(name)".cpp -o "$(name)"
-}
-c(){
-	`[ a"$1" = a-d ] && echo valgrind` ./"$(name)"
-}
-run(){
-	b "$@" && c "$@"
-}
-e(){
-	b "$@"
-	for i in "$(name)".in*; do echo $i:; c "$@" < $i; done
+	fl=-O2\ -fsanitize=address
+	[ a"$1" = a-d ] && fl=-g && v=valgrind
+	g++ -std=c++11 -lm -Wall -Wno-sign-compare -Wshadow $fl `t`.cpp -o `t`
+	for i in `t`.in*; do echo $i:; $v ./`t` < $i; done
+}
+n(){
```

Comment:
Maybe having a function for making a template is not necessary if we just create templates for all tasks at the beginning of the contest.